### DISCUSSION TITLE
build falid

### DISCUSSION THREAD

**Discussion by tangfang-sy** [osmandapp/OsmAnd#11702] [Q&A]
🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑

Hi sir，

I am a new user of OsmAnd. thers are some errors when I build omsand.

I use "../gradlew --refresh-dependencies clean assembleFullLegacyFatDebug" to build.
And the source code was downloaded by repo, branch is r3.9.

Following log outputed:
[
Configure project :OsmAnd
androidApis
androidTestUtil
archives
compile
coreLibraryDesugaring
default
lintChecks
lintClassPath
lintPublish
testCompile
Validating translations
WARNING:: API ‘variant.getJavaCompiler()’ is obsolete and has been replaced with ‘variant.getJavaCompileProvider()’.
It will be removed in version 7.0 of the Android Gradle plugin.
For more information, see https://d.android.com/r/tools/task-configuration-avoidance.
To determine what is calling variant.getJavaCompiler(), use -Pandroid.debug.obsoleteApi=true on the command line to display more information.

Configure project :OsmAnd-telegram
Warning: The ‘kotlin-android-extensions’ Gradle plugin is deprecated. Please use this migration guide (https://goo.gle/kotlin-android-extensions-deprecation) to start working with View Binding (https://developer.android.com/topic/libraries/view-binding) and the ‘kotlin-parcelize’ plugin.
WARNING:: API ‘variant.getPreBuild()’ is obsolete and has been replaced with ‘variant.getPreBuildProvider()’.
It will be removed in version 7.0 of the Android Gradle plugin.
For more information, see https://d.android.com/r/tools/task-configuration-avoidance.
To determine what is calling variant.getPreBuild(), use -Pandroid.debug.obsoleteApi=true on the command line to display more information.
WARNING:: Please remove usages of jcenter() Maven repository from your build scripts and migrate your build to other Maven repositories.
This repository is deprecated and it will be shut down in the future.
See http://developer.android.com/r/tools/jcenter-end-of-service for more information.
Currently detected usages in: root project ‘android’, project ‘:OsmAnd’, project ‘:OsmAnd-api’, …

FAILURE: Build failed with an exception.

What went wrong:
Task ‘assembleFullLegacyFatDebug’ not found in project ‘:OsmAnd’.

Try:
Run gradlew tasks to get a list of available tasks. Run with –stacktrace option to get the stack trace. Run with –info or –debug option to get more log output. Run with –scan to get full insights.

Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use ‘–warning-mode all’ to show the individual deprecation warnings.
See https://docs.gradle.org/6.7.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 42s
]

What should I do ?
Thank you!

🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑🛑

**Comment by tangfang-sy** [comment]
I updated my JDK, then now the error is blow:

[
In file included from jni/../../../core-legacy/targets/android/OsmAndCore/../../../native/src/commonRendering.cpp:1:
jni/../../../core-legacy/targets/android/OsmAndCore/../../../native/src/commonRendering.h:4:10: fatal error: ‘SkBitmap.h’ file not found

#include <SkBitmap.h>
^~
In file included from jni/../../../core-legacy/targets/android/OsmAndCore/../../../native/src/common.cpp:3:
jni/../../../core-legacy/targets/android/OsmAndCore/../../../native/src/commonOsmAndCore.h:5:10: fatal error: ‘SkPath.h’ file not found

#include <SkPath.h>
^~
11 error error generated.
]

It seems i missed some third party libs, but what it is? How to resolve it?

Is it SKia and qtbse in OsmandApp ? How to install them and thier env var?

**Reply by vshcherb** [reply]
that's tricky they are part of https://github.com/osmandapp/OsmAnd-core-legacy (https://github.com/osmandapp/OsmAnd-core-legacy/blob/master/android-configure.sh)
